Type declaration incomplete for Location.query

### Version
2.7.0

### Reproduction link
[https://jsfiddle.net/50wL7mdz/82013/](https://jsfiddle.net/50wL7mdz/82013/) (This doesn't actually work, but shows the code you'd need locally to repro the bug)

### Steps to reproduce
0. Note that JSFiddle doesn't appear to give Typescript compiler errors, so the repro linked will have to be ported into a local Typescript Vue project.

1. In a Typescript Vue project:
```
router.push({ query: { array: ["1", "2", "3"] }})
```

### What is expected?
This should compile without error 

### What is actually happening?
You get an error:
```
 error TS2345: Argument of type '{ name: string; query: QueryParams; }' is not assignable to parameter of type 'RawLocation'.
  Type '{ name: string; query: QueryParams; }' is not assignable to type 'Location'.
    Types of property 'query' are incompatible.
      Type 'QueryParams' is not assignable to type 'Dictionary<string> | undefined'.
        Type 'QueryParams' is not assignable to type 'Dictionary<string>'.
          Index signatures are incompatible.
            Type 'string | string[] | undefined' is not assignable to type 'string'.
              Type 'undefined' is not assignable to type 'string'.
```

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/types/router.d.ts
+++ b/types/router.d.ts
@@ -107,7 +107,7 @@ export interface Location {
   name?: string;
   path?: string;
   hash?: string;
-  query?: Dictionary<string>;
+  query?: Dictionary<string | string[]>;
   params?: Dictionary<string>;
   append?: boolean;
   replace?: boolean;
@@ -117,7 +117,7 @@ export interface Route {
   path: string;
   name?: string;
   hash: string;
-  query: Dictionary<string>;
+  query: Dictionary<string | string[]>;
   params: Dictionary<string>;
   fullPath: string;
   matched: RouteRecord[];

--- a/types/test/index.ts
+++ b/types/test/index.ts
@@ -105,7 +105,7 @@ const route: Route = router.currentRoute;
 const path: string = route.path;
 const name: string | undefined = route.name;
 const hash: string = route.hash;
-const query: string = route.query["foo"];
+const query: string | string[] = route.query["foo"];
 const params: string = route.params["bar"];
 const fullPath: string = route.fullPath;
 const redirectedFrom: string | undefined = route.redirectedFrom;
@@ -148,7 +148,8 @@ router.push({
     foo: "foo"
   },
   query: {
-    bar: "bar"
+    bar: "bar",
+    foo: ["foo1", "foo2"]
   },
   hash: "hash"
 });

